### PR TITLE
[Free Trial] Account Limitations for Registered Users and Courses app…

### PR DESCRIPTION
**Pull request by madnihamza1841**
**Description:**
Account Limitations for plan registered users and total courses were added and applied in LMS and Studio. 
Error messages were also shown.

**JIRA:**
https://edlyio.atlassian.net/browse/EDLY-4750

### CODE REVIEW

File: cms/djangoapps/contentstore/views/course.py [hunk 1]
```diff
@@ -931,6 +937,22 @@ def _create_or_rerun_course(request):
         )
 
 
+def _courses_limit_reached_for_plan(request):
```
Comment (taimoor-ahmed-1) [inline cms/djangoapps/contentstore/views/course.py:940]: kindly move method to `edly` app

File: openedx/core/djangoapps/user_api/accounts/__init__.py [hunk 2]
```diff
@@ -63,6 +63,10 @@
     u"It looks like {username} belongs to an existing account. "
     u"Try again with a different username."
 )
+PLAN_USER_LIMIT_REACHED_MSG = _(
```
Comment (taimoor-ahmed-1) [inline openedx/core/djangoapps/user_api/accounts/__init__.py:66]: move to edly constants

File: openedx/core/djangoapps/user_authn/views/register.py [hunk 3]
```diff
@@ -501,6 +508,29 @@ def post(self, request):
         set_logged_in_cookies(request, response, user)
         return response
 
+    def _registered_user_limit_reached_for_plan(self, request):
```
Comment (taimoor-ahmed-1) [inline openedx/core/djangoapps/user_authn/views/register.py:511]: move to `edly` app.

**Review by taimoor-ahmed-1 — CHANGES_REQUESTED** [review]
Basic idea is to have minimal changes in core apps, please move methods in our custom `openedx/features/edly` app.

File: openedx/features/edly/validators.py [hunk 4]
```diff
@@ -68,3 +80,46 @@ def is_edly_user_allowed_to_login_with_social_auth(request, user):
             return False
 
     return True
+
+
+def is_courses_limit_reached_for_plan():
+    """
+        Checks if the limit for the current site for number of courses is reached.
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/validators.py:87]: remove indent

File: cms/djangoapps/contentstore/views/course.py [hunk 5]
```diff
@@ -75,6 +75,7 @@
 from openedx.features.content_type_gating.partitions import CONTENT_TYPE_GATING_SCHEME
 from openedx.features.course_experience.waffle import ENABLE_COURSE_ABOUT_SIDEBAR_HTML
 from openedx.features.course_experience.waffle import waffle as course_experience_waffle
+from openedx.features.edly.validators import is_courses_limit_reached_for_plan
 from openedx.features.edly.utils import filter_courses_based_on_org, get_edx_org_from_cookie, get_enabled_organizations
```
Comment (taimoor-ahmed-1) [inline cms/djangoapps/contentstore/views/course.py:79]: switch positions of imports

**Review by taimoor-ahmed-1 — CHANGES_REQUESTED** [review]

File: openedx/features/edly/validators.py (hunk 4 above)
Comment (madnihamza1841) [reply openedx/features/edly/validators.py:87]: done

File: cms/djangoapps/contentstore/views/course.py (hunk 5 above)
Comment (madnihamza1841) [reply cms/djangoapps/contentstore/views/course.py]: done

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM 👍

**Review by marslanabdulrauf — APPROVED** [review]

**Review by faatehsultan — APPROVED** [review]

**Pull request merged**